### DISCUSSION TITLE
Lottery-On-VRFConsumerBaseV2

### DISCUSSION THREAD

**Discussion by Niferu** [smartcontractkit/full-blockchain-solidity-course-py#1791] [Show and tell]
Hello Everyone!

Since I have finished my Chainlink Hackathon 2022 project and currently waiting for results I wanted some new project and challenge to complete in my coding and blockchain journey, which I have started few months ago with this course. As we know VRF_V1 is pretty old, which is providing some confusion among people in this course, so here I'm to help you!

I present you lottery fully built from scratch using VRF_V2, with fully working running scripts for local network and testnet, where both works same and even on local network you will receive random number!

**Link to my repo:**
https://github.com/Neftyr/Lottery-On-VRFConsumerBaseV2

I will try to clear run script a bit more later on to avoid duplicating same functions etc.. I will also work on tests for all contract functions. Feel free to give me your thoughts and let me know if it was helpful for you!

Special thanks to @PatrickAlphaC for courses he provide disinterestedly and @cromewar for great support in this course!

**Comment by cromewar** [comment]
This is amazing, thank I'm pinning this.

**Comment by Niferu** [comment]
I have already implemented all integration and unit tests!

If anyone has questions feel free to ask, I will try to clarify everything ;)

Enjoy playing with lottery on VRF_V2!

**Comment by 0xusmanf** [comment]
Thank you so much @Neftyr you just saved my day again.
